frog/imports/ui/Editor/ActivityType.jsx

```diff
@@ -0,0 +1,212 @@
+      this.setState({listQuiz:allQuiz, nbQuiz: this.state.nbQuiz + 1});
+    }
+
+  }
+
+  //Once requested, this component generates the sub-form answer
+  generateAnswers() {
+    switch (this.props.type) {
+      case this.props.LECTURE_TYPE:
+      return ({path: this.state.lectureURL});
+
+      case this.props.VIDEO_TYPE:
+      return ({url: this.state.videoURL});
+
+      case this.props.QUIZ_TYPE:
+      return (this.generateQuizAnswers());
+
+      default:
+      return("");
+    }
+  }
+
+  //Once requested, this component generates the form answer of all quizzes
+  generateQuizAnswers() {
+
+    var questions = this.state.listQuiz.map((ref) => {
+      var quiz = (this.refs[ref]);
+      return quiz.generateQuiz();
+    });
+
+    return questions;
+  }
+
+  renderAllQuiz() {
+    return(
+      <div>
+      {this.state.listQuiz.map((ref, i) =>
+				<Quiz ref={ref} key={ref} id={i}
+					callBack={this.countSelected.bind(this)}/>)}
+
+      </div>
+      );
+  }
+
+	//Keep only the unselected quizzes
+	cleanQuizList(event) {
+		event.preventDefault();
+
+		var newListQuiz = this.state.listQuiz.filter((ref) =>{
+			var quiz = (this.refs[ref]);
+			return !quiz.isSelected();
+		});
+
+		this.setState({listQuiz: newListQuiz, nbSelected: 0});
+	}
+
+  handleVideoOrURLChange(event) {
+    event.preventDefault();
+    this.setState({lectureURL:event.target.value.trim()});
+    this.setState({videoURL:event.target.value.trim()});
+  }
+
+  //Renders according to the type given in the props
+  renderType() {
+    switch (this.props.type) {
+      case this.props.LECTURE_TYPE:
+      return (
+        <div>
+	        <input
+		        type="text"
+		        ref ="lectureURL"
+		        placeholder="Enter the path"
+		        onChange={this.handleVideoOrURLChange.bind(this)}
+		        onSubmit={this.handleVideoOrURLChange.bind(this)} /><br/>
+        </div>
+        );
+
+      case this.props.VIDEO_TYPE:
+      return (
+        <div>
+	        <input
+		        type="text"
+		        ref ="videoPath"
+		        placeholder="Enter the Youtube URL"
+		        onChange={this.handleVideoOrURLChange.bind(this)}
+		        onSubmit={this.handleVideoOrURLChange.bind(this)} /><br/>
+        </div>
+        );
+
+      case this.props.QUIZ_TYPE:
+      return (
+        <div >
+	        <button
+		        type="submit"
+		        onClick={this.createQuiz.bind(this)}
+		        disabled={this.tooManyQuiz()}>Create new Question</button>
+		        {this.renderAllQuiz()}
+					<button
+						type="delete"
+						onClick={this.cleanQuizList.bind(this)}
+						disabled={!this.enoughQuiz()}>
+						Delete {this.state.nbSelected} selected
+						Question{this.state.nbSelected < 2 ? '' : 's'}</button>
+        </div>
+        );
+
+      default:
+      return("");
+    }
+
+  }
+
+  render() {
+    return(
+      <div>
+      {this.renderType()}
+      </div>
+      )
+  }
+
+
+}
```

Comment:
Some lines in this file `frog/imports/ui/Editor/ActivityType.jsx` have tabs instead of spaces for tabulations. 

EDIT: Same in  `frog/imports/ui/Editor/Editor.jsx`